Edit IC5 role definition (#1153)
sourcegraph/handbook: content/product-engineering/engineering/career-development/framework.md

    <!-- IC4 -->
    <tr>
      <th id="ic4" scope="row" rowspan="3" class="level"><a class="anchor" href="#ic4"></a><abbr title="Individual Contributor">IC</abbr>4</th>
      <td colspan="3" class="level-summary">
        A particularly experienced, impactful contributor.<br>

>         IC4 is a terminal level: the team can have an unbounded number of IC4 engineers.

Reviewer: I am not sure people will know what "terminal level" means. Can we not use that phrase and instead inline a definition of what we mean?

Reviewer 2: The proposed change here actually makes it sound different than the definition I expected. To me, it means: a level at which we don't expect every engineer to advance beyond. That seems orthogonal to the question of whether the number of engineers at that level is bounded.

I also have trouble seeing how there is not an element of scarcity to IC4 as well, at least given the definitions that we have. E.g., "High-quality technical decision making, leading team-sized tasks that affect one or more complex systems or mission-critical areas" — is it really possible to have 6 people on one team with this kind of impact?

Reviewer 3: > is it really possible to have 6 people on one team with this kind of impact?
Probably not, but also I don't think it's a good idea to have six IC4s on one team. That feels like a needless concentration of expertise that will leave those engineers under-utilized. (Unless the team has 20 people on it, I guess.)

What about something like this: "IC4 is the highest level that an engineer can get to based solely on their impact. Beyond IC4, both impact and business need determine the availability of those positions."

(It's not perfect, but I think it's a bit closer?)

Reviewer 2: @chrispine Sounds good to me!

Reviewer: yeah that makes sense

Author: ```suggestion
        IC4 is the highest level that an engineer can get to based solely on their impact. Beyond IC4, both impact and business need determine the availability of those positions. 
```

    <tr>
      <th id="ic5" scope="row" rowspan="3" class="level"><a class="anchor" href="#ic5"></a><abbr title="Individual Contributor">IC</abbr>5</th>
      <td colspan="3" class="level-summary">
        IC5 is a different role from levels 1-4<br>
        What impact looks like at this level varies between individuals.<br>
        We recommend <a href="https://drive.google.com/drive/folders/1fnw5gmlLDDZb9wDodY3KfcTaM3OOQL7G" target="_blank" rel="noopener noreferrer">our company’s copy of “Staff Engineer” by Will Larson</a> as reading material.
      </td>
    </tr>
    <tr>
      <td>

>         Rather than being defined through proficiency, execution and teamwork IC5 is differentiated from levels 1-4 by a specific <strong>business need</strong> for the role, and an associated set of <strong>expectations</strong>.

Reviewer 3: ```suggestion
        Rather than being defined through proficiency, execution and teamwork, IC5 is differentiated from levels 1-4 by a specific <strong>business need</strong> for the role, and an associated set of <strong>expectations</strong>.
```

        <ul>

>           <li><strong>Business need</strong>: in some cases the business will need IC to and solve cross-cutting technical problems at the org or company level, willing to navigate across orgs as new problems are identified. In others, it will need a deep technical expert for a specific area, capable of devising solutions to highly complex and specialized technical problems.</li>

Reviewer 3: ```suggestion
          <li><strong>Business need</strong>: in some cases the business will need an IC to and solve cross-cutting technical problems at the org or company level, willing to navigate across orgs as new problems are identified. In others, it will need a deep technical expert for a specific area, capable of devising solutions to highly complex and specialized technical problems.</li>
```

          <li><strong>Expectations</strong>: for some IC5s, expectations might include participating in engineering calibration, because they are expected to define hiring needs for their team. For others, it might be building and owning a roadmap.</li>
        </ul>

>         The business need and associated expectations are unique to each IC5 position. IC5 engineers are publicly identified as such, and the business need and expectations associated with their role are documented in the handbook.

Reviewer 4: Does it make sense to mention that IC5 will have some element of scarcity?  It's not a lead element but we may want to highlight that IC4 is the terminal level and that IC5 may be blocking until a need occurs and make that very clear.

Author: @scalabilitysolved what do you think of https://github.com/sourcegraph/handbook/pull/1153/commits/7e46402130b4974eb7155eaee578f93c87dac2ae?

Reviewer 4: Sounds good :+1:

    </tr>
    <!-- IC5 -->
    <tr>
      <th id="ic5" scope="row" rowspan="2" class="level"><a class="anchor" href="#ic5"></a><abbr title="Individual Contributor">IC</abbr>5</th>
      <td colspan="3" class="level-summary">
        IC5 is a different role from levels 1-4.<br>
        What impact looks like at this level varies between individuals.<br>

>         We recommend <a href="https://drive.google.com/drive/folders/1fnw5gmlLDDZb9wDodY3KfcTaM3OOQL7G" target="_blank" rel="noopener noreferrer">our company’s copy of “Staff Engineer” by Will Larson</a> as reading material.

Reviewer 5: Having a reference to the Staff Engineer book here might be misleading in our attempt not to box all these roles in as "staff"-type roles

      </td>
    </tr>
    <tr>
      <td colspan="3" class="tbd">
        <p>
          Rather than being defined through proficiency, execution and teamwork, IC5 is differentiated from levels 1-4 by a specific <strong>business need</strong> for the role, and an associated set of <strong>expectations</strong>.
        </p>
        <ul>

>           <li><strong>Business need</strong>: in some cases the business will need an IC to and solve cross-cutting technical problems at the org or company level, willing to navigate across orgs as new problems are identified. In others, it will need a deep technical expert for a specific area, capable of devising solutions to highly complex and specialized technical problems.</li>

Reviewer 5: ```suggestion
          <li><strong>Business need</strong>: in some cases, the business will need an IC to solve cross-cutting technical problems at the org or company level and be willing to navigate across orgs as new problems are identified. In others, it will need a deep technical expert for a specific area, capable of devising solutions to highly complex and specialized technical problems and leading the implementation of those solutions.</li>
```

>           <li><strong>Expectations</strong>: for some IC5s, expectations might include participating in engineering calibration, because they are expected to define hiring needs for their team. For others, it might be building and owning a roadmap.</li>

Reviewer 6: This seemed like an odd example to call out.  Do we see a scenario where manager/director aren't the hiring manager?

        </ul>
        <p>
          The business need and associated expectations are unique to each IC5 position. IC5 engineers are publicly identified as such, and the business need and expectations associated with their role are documented in the handbook.
        </p>
        <p>

>           Unlike IC4, IC5 is not a terminal level. Transition to the IC5 level is contingent on the existence of a business need for the role.

Reviewer 6: When I first read this, it makes it sound like IC5 role might disappear if the need isn't there anymore, not sure if that's the intent
